atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                        continue
+                    elif len(self._system_info[sec_key][key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of particle attributes not supported. ' + key + ' values will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][key] = [self._system_info[sec_key][key]] * n_frames
+
+            # TODO Should we extend this to pick up additional attributes in the particles group? Or require that we follow the H5MD schema strictly?
+
+            # get the system box quantities
+            box = self._data_parser.hdf5_getter(particles_group, 'box')
+            if box is None:
+                return
+
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
+            for box_key, sec_key in box_attributes.items():
+                value = self._data_parser.hdf5_attr_getter(box, self._nomad_to_box_group_map[box_key], path=None)
+                if value is not None:
+                    self._system_info[sec_key][box_key] = [value] * n_frames
+
+            box_keys = {'lattice_vectors': 'system'}
+            for box_key, sec_key in box_keys.items():
+                value = self._data_parser.hdf5_getter(box, self._nomad_to_box_group_map[box_key])
+                if value is not None:
+                    self._system_info[sec_key][box_key] = value
+                else:
+                    continue
+
+                if type(self._system_info[sec_key][box_key]) == h5py.Group:
+                    self._system_info[sec_key][box_key] = self._data_parser.hdf5_getter(self._system_info[sec_key][box_key], 'value')
+                    if self._system_info[sec_key][box_key] is None:
+                        continue
+                    elif len(self._system_info[sec_key][box_key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of box vectors and positions is not supported. ' + box_key + ' values will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][box_key] = [self._system_info[sec_key][box_key]] * n_frames
+        return self._system_info
+
+    @property
+    def observable_info(self):
+        if self._observable_info is None:
+            observables_group = self.h5md_groups.get('observables')
+            if observables_group is None:
+                return
+
+            self._observable_info = {
+                'configurational': {},
+                'ensemble_average': {},
+                'correlation_function': {}
+            }
+
+            def get_observable_paths(observable_group, current_path, paths):
```

Comment:
sorry, I can't quite figure this one out. I don't understand where paths should be defined then, since I am calling the function recursively? Are you saying to completely remove paths as a function argument? Or just somehow not pass it when I call it within the function?

Reply:
you can initialize paths = [] in ll330 and then do paths.extend(get_observable_paths(observable, current_path + path)) in l335

Reply:
aha, ok thank you. I guess it was just intuitive to me for some reason what would happen with extend...thanks! done.